sdk/database/newdbplugin/grpc_server_test.go

```diff
@@ -0,0 +1,692 @@
+			expectErr:  false,
+			expectCode: codes.OK,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			g := gRPCServer{
+				impl: test.db,
+			}
+
+			// Context doesn't need to timeout since this is just passed through
+			ctx := context.Background()
+
+			resp, err := g.NewUser(ctx, test.req)
+			if test.expectErr && err == nil {
+				t.Fatalf("err expected, got nil")
+			}
+			if !test.expectErr && err != nil {
+				t.Fatalf("no error expected, got: %s", err)
+			}
+
+			actualCode := status.Code(err)
+			if actualCode != test.expectCode {
+				t.Fatalf("Actual code: %s Expected code: %s", actualCode, test.expectCode)
+			}
+
+			if !reflect.DeepEqual(resp, test.expectedResp) {
+				t.Fatalf("Actual response: %#v\nExpected response: %#v", resp, test.expectedResp)
+			}
+		})
+	}
+}
+
+func TestGRPCServer_UpdateUser(t *testing.T) {
+	type testCase struct {
+		db           Database
+		req          *proto.UpdateUserRequest
+		expectedResp *proto.UpdateUserResponse
+		expectErr    bool
+		expectCode   codes.Code
+	}
+
+	tests := map[string]testCase{
+		"missing username": {
+			db:           fakeDatabase{},
+			req:          &proto.UpdateUserRequest{},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    true,
+			expectCode:   codes.InvalidArgument,
+		},
+		"missing changes": {
+			db: fakeDatabase{},
+			req: &proto.UpdateUserRequest{
+				Username: "someuser",
+			},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    true,
+			expectCode:   codes.InvalidArgument,
+		},
+		"database error": {
+			db: fakeDatabase{
+				updateUserErr: errors.New("update user error"),
+			},
+			req: &proto.UpdateUserRequest{
+				Username: "someuser",
+				Password: &proto.ChangePassword{
+					NewPassword: "90ughaino",
+				},
+			},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    true,
+			expectCode:   codes.Internal,
+		},
+		"bad expiration date": {
+			db: fakeDatabase{},
+			req: &proto.UpdateUserRequest{
+				Username: "someuser",
+				Expiration: &proto.ChangeExpiration{
+					NewExpiration: &timestamp.Timestamp{
+						// Before minValidSeconds in ptypes package
+						Seconds: invalidExpiration.Unix(),
+					},
+				},
+			},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    true,
+			expectCode:   codes.InvalidArgument,
+		},
+		"change password happy path": {
+			db: fakeDatabase{},
+			req: &proto.UpdateUserRequest{
+				Username: "someuser",
+				Password: &proto.ChangePassword{
+					NewPassword: "90ughaino",
+				},
+			},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    false,
+			expectCode:   codes.OK,
+		},
+		"change expiration happy path": {
+			db: fakeDatabase{},
+			req: &proto.UpdateUserRequest{
+				Username: "someuser",
+				Expiration: &proto.ChangeExpiration{
+					NewExpiration: ptypes.TimestampNow(),
+				},
+			},
+			expectedResp: &proto.UpdateUserResponse{},
+			expectErr:    false,
+			expectCode:   codes.OK,
+		},
+	}
```

Comment:
I don't think it would be valuable since the statements aren't actually used anywhere. The gRPC client is faked, so the statements wouldn't go anywhere.

Reply:
Ok, though I'm wondering if some of the conversion functions from struct <--> proto could use tests, since they seem to be ignoring some of the fields?

Reply:
I added a new series of tests in `conversions_test.go` that helps ensure we don't miss any fields.